examples/Microphysics/ex_2_Kessler.jl

```diff
@@ -0,0 +1,421 @@
+# The test is based on a modelling set-up designed for the
+# 8th International Cloud Modelling Workshop
+# (ICMW, Muhlbauer et al., 2013, case 1, doi:10.1175/BAMS-D-12-00188.1)
+#
+# See chapter 2 in Arabas et al 2015 for setup details:
+#@Article{gmd-8-1677-2015,
+#AUTHOR = {Arabas, S. and Jaruga, A. and Pawlowska, H. and Grabowski, W. W.},
+#TITLE = {libcloudph++ 1.0: a single-moment bulk, double-moment bulk, and particle-based warm-rain microphysics library in C++},
+#JOURNAL = {Geoscientific Model Development},
+#VOLUME = {8},
+#YEAR = {2015},
+#NUMBER = {6},
+#PAGES = {1677--1707},
+#URL = {https://www.geosci-model-dev.net/8/1677/2015/},
+#DOI = {10.5194/gmd-8-1677-2015}
+#}
+
+using MPI
+using CLIMA.Topologies
+using CLIMA.Grids
+using CLIMA.DGBalanceLawDiscretizations
+using CLIMA.DGBalanceLawDiscretizations.NumericalFluxes
+using CLIMA.MPIStateArrays
+using CLIMA.LowStorageRungeKuttaMethod
+using CLIMA.ODESolvers
+using CLIMA.GenericCallbacks
+using CLIMA.Vtk
+
+using LinearAlgebra
+using StaticArrays
+using Printf
+
+using CLIMA.PlanetParameters
+using CLIMA.MoistThermodynamics
+using CLIMA.Microphysics
+
+@static if Base.find_package("CuArrays") !== nothing
+  using CUDAdrv
+  using CUDAnative
+  using CuArrays
+  const ArrayTypes = VERSION >= v"1.2-pre.25" ? (Array, CuArray) : (Array,)
+else
+  const ArrayTypes = (Array, )
+end
+
+const _nstate = 7
+const _ρ, _ρu, _ρw, _ρe_tot, _ρq_tot, _ρq_liq, _ρq_rai = 1:_nstate
+const stateid = (ρ_id = _ρ, ρu_id = _ρu, ρw_id = _ρw, ρe_tot_id = _ρe_tot,
+                 ρq_tot_id = _ρq_tot, ρq_liq_id = _ρq_liq, ρq_rai_id = _ρq_rai)
+const statenames = ("ρ", "ρu", "ρw", "ρe_tot", "ρq_tot", "ρq_liq", "ρq_rai")
+
+const _nauxcstate = 3
+const _c_z, _c_x, _c_p = 1:_nauxcstate
+
+
+# preflux computation
+@inline function preflux(Q, _...)
+  DFloat = eltype(Q)
+  @inbounds begin
+    # unpack all the state variables
+    ρ, ρu, ρw, ρq_tot, ρq_liq, ρq_rai, ρe_tot = Q[_ρ], Q[_ρu], Q[_ρw],
+                                                Q[_ρq_tot], Q[_ρq_liq],
+                                                Q[_ρq_rai], Q[_ρe_tot]
+    u, w, q_tot, q_liq, q_rai, e_tot = ρu / ρ, ρw / ρ,
+                                       ρq_tot / ρ, ρq_liq / ρ,
+                                       ρq_rai / ρ, ρe_tot / ρ
+
+    DF = eltype(ρ)
+    if(q_rai >= DF(0))
+      # compute rain fall speed
+      rain_w = terminal_velocity(q_rai, ρ)
```

Comment:
As we discussed over slack today.  You need to make the following change because you are zeroing out `rain_w` in the `eulerflux!` function.

```suggestion
      rain_w = DF(0)
```

Reply:
Yes. Just added. Testing now

Reply:
Thank you!